File: code/java/ch20/practice/Warmup03MaxAreaOfIsland.java
package ch20.practice;

import java.util.*;

/**
 * Warmup 3: Max Area of Island
 * Chapter 20: Graphs II — Real Problems
 *
 * PROBLEM: Given an m x n binary grid, return the area of the largest island.
 *
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */
public class Warmup03MaxAreaOfIsland {
    public static int solve(int[][] grid) {
        // TODO: Replace this with your solution
        return 0;
    }

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int rows = scan.nextInt(), cols = scan.nextInt();
        int[][] grid = new int[rows][cols];
        for (int i = 0; i < rows; i++)
            for (int j = 0; j < cols; j++)
                grid[i][j] = scan.nextInt();
        System.out.println(solve(grid));
        scan.close();
    }
}

File: code/python/ch20/practice/warmup_03_max_area_of_island.py
"""
Warmup 3: Max Area of Island
============================================
Chapter 20: Graphs II -- Real Problems

PROBLEM
-------
Given an m x n grid of 0s and 1s, return the maximum area of an island.
An island's area is the number of 1s in a 4-directionally connected group.

EXAMPLES
--------
>>> solve([[0,0,1,0,0],[0,0,1,0,0],[0,1,1,0,1],[0,0,1,0,0]])
5

CONSTRAINTS
-----------
- 1 <= m, n <= 50
- grid[i][j] is 0 or 1
"""


def solve(grid: list[list[int]]) -> int:
    """Return the maximum area of an island."""
    pass


# -- Do not change anything below this line ----------------------------------
if __name__ == "__main__":
    first_line = input().split()
    rows, cols = int(first_line[0]), int(first_line[1])
    grid = []
    for _ in range(rows):
        grid.append(list(map(int, input().split())))
    print(solve(grid))
